src/App.js

```diff
@@ -1,26 +1,114 @@
 import React from 'react';
-import logo from './logo.svg';
+import Bucket from './Bucket';
 import './App.css';
+import { Container } from 'reactstrap';
+import {
+  InputGroup,
+  InputGroupAddon,
+  InputGroupText,
+  Col,
+  Row,
+  Label,
+  Input
+} from 'reactstrap';
 
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+class App extends React.Component {
+  state = {
+    income: 0,
+    buckets: {
+      "Fixed Costs": 50,
```

Comment:
Why is this "Fixed Costs" and not `FixedCosts` ?

Reply:
It appears that the keys were left "human readable" here so the  `renderBuckets` function could use array like iteration over `state.buckets` via `Object.keys(buckets).map(...)` which is probably fine for now.

If the buckets ever need to be extended to include any details beyond "name" and "percentage" an array of objects would be more suitable.

eg ` state.buckets = [ {name: "Fixed Costs" percentage: 50, color: "red"}, ...]`